pkg/login/social/connectors/azuread_oauth.go

```diff
@@ -167,6 +167,13 @@ func (s *SocialAzureAD) Validate(ctx context.Context, settings ssoModels.SSOSett
 }
 
 func (s *SocialAzureAD) Reload(ctx context.Context, settings ssoModels.SSOSettings) error {
+	info, err := CreateOAuthInfoFromKeyValues(settings.Settings)
+	if err != nil {
+		return fmt.Errorf("SSO settings map cannot be converted to OAuthInfo: %v", err)
+	}
+
+	s.info = info
```

Comment:
we need to use a mutex when reading and writing config because this will happen during runtime right?

Reply:
yes, good point, I'll use a mutex.

Reply:
FYI we also need it for every read path